tests/backup/backup_scale_test.go

```diff
@@ -210,3 +210,148 @@ var _ = Describe("{MultipleBackupLocationWithSameEndpoint}", func() {
 		CleanupCloudSettingsAndClusters(backupLocationMap, cloudCredName, cloudCredUID, ctx)
 	})
 })
+
+// This TC takes backup of 50 volumes and performs restore
+var _ = Describe("{ValidateFiftyVolumeBackups}", func() {
+	var (
+		scheduledAppContexts []*scheduler.Context
+		sourceClusterUid     string
+		backupLocationMap    map[string]string
+		cloudAccountName     string
+		bkpLocationName      string
+		cloudCredUID         string
+		backupLocationUID    string
+		currentBackupName    string
+		namespace            string
+		backupNameList       []string
+		restoreNames         []string
+		preRuleName          string
+		postRuleName         string
+		preRuleUid           string
+		postRuleUid          string
+		providers            = GetBackupProviders()
+		numberOfVolumes      = 50
+	)
+
+	JustBeforeEach(func() {
+		StartPxBackupTorpedoTest("ValidateFiftyVolumeBackups", "To verify backup of 50 volumes and performs restore", nil, 55816, Sabrarhussaini, Q1FY25)
+		backupLocationMap = make(map[string]string)
+		log.InfoD("scheduling applications")
+		appList := Inst().AppList
+		defer func() {
+			Inst().AppList = appList
+		}()
+		namespace = fmt.Sprintf("multiple-app-ns-%s", RandomString(6))
+		Inst().AppList = []string{"postgres-backup-with-suffix"}
+		for i := 0; i < numberOfVolumes; i++ {
+			Inst().CustomAppConfig["postgres-backup-with-suffix"] = scheduler.AppConfig{
+				Suffix: RandomString(8),
+			}
+			err := Inst().S.RescanSpecs(Inst().SpecDir, Inst().V.String())
```

Comment:
I think we can put this outside the for loop. There is no need to rescan specs every iteration.

Reply:
@kphalgun-px please correct me if I'm wrong, we are rescanning because we are updating the suffix for every iteration and to make sure that the suffix is update, we will have to rescan, right?

Reply:
My bad. I think I figured out a better way to do this. Will add a comment soon.